fendermint/fendermint/vm/interpreter/src/fvm/state/query.rs

```diff
@@ -204,6 +205,25 @@ where
         &self.state_params
     }
 
+    /// Returns the registry of built-in actors as enrolled in the System actor.
+    pub async fn builtin_actors(self) -> anyhow::Result<(Self, Vec<(String, Cid)>)> {
+        let (s, sys_state) = {
+            let (s, state) = self.actor_state(&SYSTEM_ACTOR_ADDR).await?;
+            (s, state.ok_or(anyhow!("no system actor"))?.1)
+        };
+        let state: State = s
+            .store
+            .get_cbor(&sys_state.code)
```

Comment:
```suggestion
            .get_cbor(&sys_state.state)
```

I think you're getting the wrong piece of data there. The [system state](https://github.com/consensus-shipyard/ipc-monorepo/blob/03b745c7b64c2cb7888fbffd75d36218f02c02ae/fendermint/fendermint/vm/interpreter/src/fvm/genesis.rs#L151-L153) is indeed the manifest but it's [not stored in the code field](https://github.com/consensus-shipyard/ipc-monorepo/blob/03b745c7b64c2cb7888fbffd75d36218f02c02ae/fendermint/fendermint/vm/interpreter/src/fvm/state/genesis.rs#L171-L181). 

It looks like a typo but got me confused if there is a trick I forgot. This is where using `TCid` would be useful.